src/main/java/org/osiam/resources/controller/MeController.java

```diff
@@ -23,186 +23,61 @@
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.oauth2.provider.OAuth2Authentication;
 import org.springframework.security.oauth2.provider.token.ResourceServerTokenServices;
 import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.RequestHeader;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RestController;
-
-import javax.servlet.http.HttpServletRequest;
-import java.util.Set;
+import org.springframework.web.util.UriComponentsBuilder;
 
 /**
  * This Controller is used for getting information about the user who initialised the access_token.
  */
 @RestController
-@RequestMapping(value = "/me")
+@RequestMapping(value = "/Me")
 @Transactional
-public class MeController {
+public class MeController extends ResourceController<User> {
 
     private final ResourceServerTokenServices resourceServerTokenServices;
-    private final UserDao userDao;
 
     @Autowired
-    public MeController(ResourceServerTokenServices resourceServerTokenServices, UserDao userDao) {
+    public MeController(ResourceServerTokenServices resourceServerTokenServices) {
         this.resourceServerTokenServices = resourceServerTokenServices;
-        this.userDao = userDao;
     }
 
-    /**
-     * This method is used to get information about the user who initialised the authorization process.
-     * <p/>
-     * The result should be in json format and look like:
-     * <p/>
-     * {
-     * "id": "73821979327912",
-     * "name": "Arthur Dent",
-     * "first_name": "Arthur",
-     * "last_name": "Dent",
-     * "link": "https://www.facebook.com/arthur.dent.167",
-     * "username": "arthur.dent.167",
-     * "gender": "male",
-     * "email": "arthur@dent.de",
-     * "timezone": 2,
-     * "locale": "en_US",
-     * "verified": true,
-     * "updated_time": "2012-08-20T08:03:30+0000"
-     * }
-     * <p/>
-     * if some information are not available then ... will happen.
-     *
-     * @return an object to represent the json format.
-     */
-    @RequestMapping(value = "/**", method = {RequestMethod.GET, RequestMethod.POST})
-    public FacebookInformationConstruct getInformation(HttpServletRequest request) {
-        String accessToken = getAccessToken(request);
+    @RequestMapping(method = RequestMethod.GET)
+    public ResponseEntity<User> getInformation(@RequestHeader("Authorization") String tokenHeader,
+                                               UriComponentsBuilder builder) {
+
+        if (Strings.isNullOrEmpty(tokenHeader)) {
+            throw new IllegalArgumentException("No access token provided!"); // This should never happen!
```

Comment:
Yes. Of course it's not possible when you [secure the endpoint](https://github.com/osiam/osiam/pull/185#issuecomment-196070708) ;)